### PR TITLE
O3-1069: Fixed the time sensitivity of date of birth in registration page to time zone

### PR DESCRIPTION
**Pull request by vasharma05**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
Before, when saving the date of birth, say 01/02/2022(dd/mm/yyyy), it was getting changed to 31/01/2022, due to a change in time in local and UTC. This is fixed, along with the following changes:
1. Separated the gender and sex fields, cleaned up the configuration file.
image
image

2. Added content switcher to toggle between entering the date of birth or the estimated age.
image
image

3. Added a max date in the calendar, i.e. the date the user is filling the form since dob can't be in the future.
image

## Screenshots
image

## Related Issue
https://issues.openmrs.org/browse/O3-1069

## Other

*None.*

### CODE REVIEW

**Comment by gracepotma** [comment]
When this is eventually merged, it would be great to do a release of esm-patient-management so that these fixes are present on dev3; then we can update o3 as well.

**Review by ibacher — COMMENTED** [review]
Thanks for this @vasharma05! I'm a little hesitant to approve this only because this makes some fairly major changes to the UI that departs a bit from the designs. In particular, I can't find anywhere else where we're using a toggle like that. There are things like this:

Screenshot 2022-02-11 at 15 26 19

E.g., inside an [OHRI form](https://app.zeplin.io/project/60d5947dd636aebbd63dce4c/screen/61bb008f0e69e45e7175e63e) which is maybe less hideous than a checkbox, but less distracting than the toggle.

Most of the other changes here are overall benefits.

**Review by ZacButko — APPROVED** [review]
Very tidy and efficient code. Thanks @vasharma05 .

**Comment by jonathandick** [comment]
@vasharma05 and updates on this?

**Comment by ZacButko** [comment]
@vasharma05 are you still going to change the toggle to a 'yes / no content switcher' like Ciaran suggested?

File: packages/esm-patient-registration-app/src/patient-registration/field/gender/gender-field.component.tsx [hunk 1]
```diff
@@ -17,28 +16,24 @@ export const GenderBirthField: React.FC = () => {
...
+        <p className="bx--label">{t('genderLabelText', 'Sex')}</p>
+        <RadioButtonGroup name="gender" orientation="vertical" onChange={setGender} valueSelected={field.value}>
+          <RadioButton id="male" labelText={t('maleLabelText', 'Male')} value="Male" />
+          <RadioButton id="female" labelText={t('femaleLabelText', 'Female')} value="Female" />
+          <RadioButton id="other" labelText={t('otherLabelText', 'Other')} value="Other" />
+          <RadioButton id="unknown" labelText={t('unknownLabelText', 'Unknown')} value="Unknown" />
+        </RadioButtonGroup>
+        {meta.touched && meta.error && (
+          <>
+            <input type="hidden" data-invalid="true" />
```
[the remark this answers is not in the document]
Comment (vasharma05) [reply packages/esm-patient-registration-app/src/patient-registration/field/gender/gender-field.component.tsx]: It was present there from before. Thanks for pointing it out, I found it redundant and removed it.

File: packages/esm-patient-registration-app/src/patient-registration/field/dob/dob.component.tsx [hunk 2]
```diff
@@ -1,65 +1,84 @@
...
-  const today = new Date();
-  const [field] = useField('birthdate');
-  const [estimated] = useField('birthdateEstimated');
-  const { setFieldValue } = useContext(PatientRegistrationContext);
-
-  const onValueChange = (ev) => {
+  const onEstimatedAgeChange = (ev) => {
     const years = +ev.target.value;
 
     if (!isNaN(years) && years < 1000 && years >= 0) {
```
[the remark this answers is not in the document]
Comment (vasharma05) [reply packages/esm-patient-registration-app/src/patient-registration/field/dob/dob.component.tsx]: Updated it @ibacher!
Thanks!

File: packages/esm-patient-registration-app/src/patient-registration/field/field.scss [hunk 3]
```diff
@@ -39,7 +39,7 @@
   justify-self: center;
 }
 
-.sexField {
+.sexField, .marginBottom01 {
```
[the remark this answers is not in the document]
Comment (vasharma05) [reply packages/esm-patient-registration-app/src/patient-registration/field/field.scss]: Thanks, @ZacButko! I have renamed it.

**Comment by vasharma05** [comment]
Hi, @jonathandick @ZacButko @ibacher, apologies for the delay and thanks for your patience.

**Pull request merged**